src/device.hpp

```diff
@@ -63,12 +63,12 @@ typedef struct _cl_device_id {
     }
 
     uint64_t memory_size() const {
-        return std::min(max_alloc_size() * 4, actual_memory_size());
+        return std::min<uint64_t>(max_alloc_size() * 4, actual_memory_size());
```

Comment:
I think it's better to change the return type of `max_alloc_size()` to `uint64_t`. The API reports a `cl_ulong` anyway which, contrary to `size_t` is guaranteed to be 64-bit.